pyvista/examples/downloads.py

```diff
@@ -10,304 +10,327 @@
 >>> mesh.plot()
 
 """
-
-from functools import partial
-import glob
 import os
 import shutil
-from typing import Union
-from urllib.request import urlretrieve
-import zipfile
+from typing import Optional, Union
+import warnings
 
 import numpy as np
+import pooch
+from pooch import Unzip
 
 import pyvista
 from pyvista import _vtk
 from pyvista.core.errors import VTKVersionError
 
-CACHE_VERSION = 2
-
-
-def _check_examples_path():
-    """Check if the examples path exists."""
-    if not pyvista.EXAMPLES_PATH:
-        raise FileNotFoundError(
-            'EXAMPLES_PATH does not exist.  Try setting the '
-            'environment variable `PYVISTA_USERDATA_PATH` '
-            'to a writable path and restarting python'
-        )
-
-
-def _cache_version(cache_version_file) -> int:
-    """Return the cache version."""
-    if os.path.isfile(cache_version_file):
-        with open(cache_version_file) as fid:
-            try:
-                return int(fid.read())
-            except:
-                pass
-    return 0
-
-
-def _verify_cache_integrity():  # pragma: no cover
-    """Verify that the version of the cache matches the expected version.
-
-    Clears cache when there is a version mismatch. This avoids any potential
-    issues with old file structures due to changed download methods.
-
-    """
-    cache_version_file = os.path.join(pyvista.EXAMPLES_PATH, 'VERSION')
-    cache_version = _cache_version(cache_version_file)
-
-    # clear with no version file or an old cache version
-    if cache_version < CACHE_VERSION:
-        delete_downloads()
-        with open(cache_version_file, 'w') as fid:
-            fid.write(str(CACHE_VERSION))
-
-
-def delete_downloads():
-    """Delete all downloaded examples to free space or update the files.
-
-    Returns
-    -------
-    bool
-        Returns ``True``.
-
-    Examples
-    --------
-    Delete all local downloads.
-
-    >>> from pyvista import examples
-    >>> examples.delete_downloads()  # doctest:+SKIP
-    True
-
-    """
-    _check_examples_path()
-    if os.path.isdir(pyvista.EXAMPLES_PATH):
-        shutil.rmtree(pyvista.EXAMPLES_PATH)
-    os.makedirs(pyvista.EXAMPLES_PATH)
-    return True
-
-
-def _decompress(filename, output_path):
-    _check_examples_path()
-    zip_ref = zipfile.ZipFile(filename, 'r')
-    zip_ref.extractall(output_path)
-    return zip_ref.close()
-
-
-def _get_vtk_file_url(filename):
-    return f'https://github.com/pyvista/vtk-data/raw/master/Data/{filename}'
+CACHE_VERSION = 3
 
+# If available, a local vtk-data instance will be used for examples
 
-def _http_request(url, progress_bar=False):  # pragma: no cover
-    """Download a file from a url using ``urlretrieve``.
+VTK_DATA_PATH: Optional[str] = None
+if 'PYVISTA_VTK_DATA' in os.environ:  # pragma: no cover
+    warnings.warn(
+        'PYVISTA_VTK_PATH has been deprecated as of 0.37.0\n\n' 'Please use PYVISTA_USERDATA_PATH'
```

Comment:
I don't think these environment variables are used the same.  One sets the path of the cache.  The other set a local git repo of `pyvista/vtk-data` used as a source for the cache.

If we keep this, we should warn that this is no longer supported and not set `PYVISTA_USERDATA_PATH`.

Reply:
Fixed in f6e5506. I've reverted to the previous behavior.